### PR TITLE
Update/lm 2592 update autotests for promises support

### PR DESCRIPTION
**Pull request by anton-locomote**
Hooks tests will be added by Max in hook task branch.

### CODE REVIEW

File: test/integration/association-hasmany-extra.js [hunk 1]
```diff
@@ -114,7 +114,11 @@ describe("hasMany extra properties", function() {
           }).catch(function(err) {
             done(err);
           });
+        }).catch(function(err) {
+          done(err);
         });
+      }).catch(function(err) {
+        done(err);
```
Comment (dxg) [inline test/integration/association-hasmany-extra.js]: Why don't we have
```js
.catch(done)
```
?

**Comment by anton-locomote** [comment]
In general didn't use usual chain, because we have situation that sometimes we have situation when we use callback -> promise -> callback -> promise in our tests. So to keep chains is quite hard in this situation, but will update tests where we can do this.

**Comment by dxg** [comment]
What happened to `test/integration/association-hasmany.js` tests? Whole file deleted?

**Comment by anton-locomote** [comment]
@dxg Async tests from those files were moved to new <filename>-async.js files. Then those tests updated according to changes from all tasks.

File: test/integration/association-hasone-async.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  });
+
+  describe("accessors Async", function () {
+    before(setup());
+
+    it("get should get the association", function () {
+      return Leaf
+        .oneAsync({ size: 14 })
+        .then(function(leaf) {
+					should.exist(leaf);
```
Comment (tttmaximttt) [inline test/integration/association-hasone-async.js]: something bad happened here. Adjust please.

File: test/integration/association-hasone-async.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      return Stalk
+        .oneAsync({ length: 20 })
+        .then(function (stalk) {
+          should.exist(stalk);
+          return [stalk, Leaf.oneAsync({ size: 14 })];
+        })
+        .spread(function (stalk, leaf) {
+          should.exist(leaf);
+          should.not.exist(leaf.stalkId);
+					return [stalk, leaf.setStalkAsync(stalk)];
```
Comment (tttmaximttt) [inline test/integration/association-hasone-async.js]: Adjust

File: test/integration/association-hasone-async.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+          should.equal(leafOne.stalkId, stalk[0][Stalk.id]);
+        });
+    });
+
+    it("remove should unassociation another instance", function () {
+      return Stalk
+        .oneAsync({ length: 20 })
+        .then(function (stalk) {
+          should.exist(stalk);
+					return Leaf.oneAsync({ size: 14 });
```
Comment (tttmaximttt) [inline test/integration/association-hasone-async.js]: adjust

File: test/integration/association-hasone-async.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+				})
+        .then(function (leaf) {
+          should.exist(leaf);
+          should.exist(leaf.stalkId);
+          return leaf.removeStalkAsync();
+				})
+        .then(function () {
+          return Leaf.oneAsync({ size: 14 });
+        })
+				.then(function (leaf) {
```
Comment (tttmaximttt) [inline test/integration/association-hasone-async.js]: adjust

File: test/integration/association-hasone-reverse-async.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+            owners[0].should.eql(ownersCopy[1]);
+            owners[1].should.eql(ownersCopy[0]);
+          }
+        });
+    });
+  });
+
+  describe("reverse find", function () {
+    before(setup());
+    it("should be able to find given an association id", function (done) {
```
Comment (tttmaximttt) [inline test/integration/association-hasone-reverse-async.js]: remove done

File: test/integration/association-hasone-reverse-async.js [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+            return [John, Person.findAsync({ pet_id: 1 })];
+          })
+          .spread(function (John, owner) {
+            should.exist(owner[0]);
+            should.equal(owner[0].name, John.name);
+          });
+       }, 3, done);
+    });
+    
+    it("should be able to find given an association instance", function (done) {
```
Comment (tttmaximttt) [inline test/integration/association-hasone-reverse-async.js]: remove done

File: test/integration/association-hasone-reverse-async.js [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+            return [John, Person.findAsync({ pet: Deco })];
+          })
+          .spread(function(John, owner){
+            should.exist(owner[1]);
+            should.equal(owner[1].name, John.name);
+          });
+      }, 3, done);
+    });
+
+    it("should be able to find given a number of association instances with a single primary key", function (done) {
```
Comment (tttmaximttt) [inline test/integration/association-hasone-reverse-async.js]: remove done

**Pull request merged**